haskell-src/Concordium/Types/Parameters.hs

```diff
@@ -687,34 +1404,184 @@ putPoolParameters PoolParametersV1{..} = do
     put _ppCapitalBound
     put _ppLeverageBound
 
-instance HashableTo Hash.Hash (PoolParameters cpv) where
+instance HashableTo Hash.Hash (PoolParameters' ppv) where
     getHash = Hash.hash . runPut . putPoolParameters
 
-instance Monad m => MHashableTo m Hash.Hash (PoolParameters cpv)
+instance Monad m => MHashableTo m Hash.Hash (PoolParameters' ppv)
 
-getPoolParameters :: forall cpv. IsChainParametersVersion cpv => Get (PoolParameters cpv)
-getPoolParameters = case chainParametersVersion @cpv of
-    SCPV0 -> PoolParametersV0 <$> get
-    SCPV1 -> PoolParametersV1 <$> get <*> get <*> get <*> get <*> get
+-- |Deserialize a 'PoolParameters'' at a given version.
+getPoolParameters :: forall ppv. SPoolParametersVersion ppv -> Get (PoolParameters' ppv)
+getPoolParameters = \case
+    SPoolParametersVersion0 -> PoolParametersV0 <$> get
+    SPoolParametersVersion1 -> PoolParametersV1 <$> get <*> get <*> get <*> get <*> get
 
-instance IsChainParametersVersion cpv => Serialize (PoolParameters cpv) where
+instance IsPoolParametersVersion ppv => Serialize (PoolParameters' ppv) where
     put = putPoolParameters
-    get = getPoolParameters
+    get = getPoolParameters sing
+
+deriving instance Eq (PoolParameters' ppv)
+deriving instance Show (PoolParameters' ppv)
+
+-- * Timeout parameters
+
+-- |Parameters controlling consensus timeouts for the consensus protocol version 2.
+data TimeoutParameters = TimeoutParameters
+    { -- |The base value for triggering a timeout.
+      tpTimeoutBase :: Duration,
+      -- |Factor for increasing the timeout. Must be greater than 1.
+      tpTimeoutIncrease :: Ratio Word64,
+      -- |Factor for decreasing the timeout. Must be between 0 and 1.
+      tpTimeoutDecrease :: Ratio Word64
+    }
+    deriving (Eq, Show)
 
-deriving instance Eq (PoolParameters cpv)
-deriving instance Show (PoolParameters cpv)
+instance Serialize TimeoutParameters where
+    put TimeoutParameters{..} = do
+        put tpTimeoutBase
+        put tpTimeoutIncrease
+        put tpTimeoutDecrease
+    get = do
+        tpTimeoutBase <- get
+        tpTimeoutIncrease <- get
```

Comment:
We are deviating from what we do in other such cases by using an instance that has implicit endianness.

Moreover in contrast to other cases with ratios, where we check that numerator and denominator are coprime we do not do so here. Which I would consider a bug. Same for decrease.

Reply:
Fix inc